Sudoku needed numbers

lumodon opened the pull request:
Spec file removed by adding to .gitignore
couldn't figure out how to remove it from git history altogether.

lumodon committed:
Adds duplicates error checking
- Also removes unneeded console log in print function

lumodon committed:
Adds assortment of error checking
- Builds lots of helper functions

pllearns commented on Sudoku/buildMap.js:
```diff
@@ -1,16 +1,19 @@
     cols: [],
-    sqrs: []
+    sqrs: [],
+    potentialNumbers: [], // TODO: Rename better. Potential numbers is all numbers that could potentially fit into a given slot. No regard to what numbers the square has already consumed.
```
Maybe rename this to `allNumbersAvailable`? Just a thought.

pllearns commented on Sudoku/errorChecking.js:
```diff
@@ -0,0 +1,81 @@
+  for(let bigSquare of squaresArr) {
+    for(let iterI in bigSquare){
+      iterI = parseInt(iterI)
+      for(let iterJ = iterI+1; iterJ < 9; iterJ++) { // What big O is this?
```
Could this be `O(n)`? Not sure either.

pllearns commented on Sudoku/errorChecking.js:
```diff
@@ -0,0 +1,81 @@
+  return false
+}
+
+const checkAnyNumbersAvailable = (rowsArr, columnsArr) => {
```
This is working, right?

pllearns reviewed (commented):
Good job, check the comments, but nothing that's stopping you from merging in my estimation.

lumodon committed:
Changes structure of data around completely
-* Adds error checking to pass test 4 i.e.:
--* Checks every square to ensure numbers that fit into that square
--  are usable and dont conflict with rows, columns, or squre itself
-* Stores numbers needed per square and numbers needed per slot
